### ISSUE TITLE
unnecessary_lambda doesn't lint

### ISSUE THREAD

**Issue by zoechi** [dart-lang/linter#1090]
```dart
            onChanged: (text) =>
                groupController.messageComposeController.textChanged(text),
```
but I think it should ;-)

**Comment by matanlurey** [comment]
This is likely not a safe assumption in Dart2. There are many cases creating these forwarding closures allows the types to be propagated in ways that they are not when a tear off occurs.
